operations/conman/Establish_a_Serial_Connection_to_NCNs.md

```diff
@@ -1,120 +1,148 @@
+
+In order to avoid losing data while monitoring a reboot or power down of a worker node,
+first follow this procedure to evacuate the target worker node of its pods.
+
+1. (`ncn-mw#`) Set the `WNODE` variable to the name of the worker node being evacuated.
+
+    Modify the following example to reflect the actual worker node number.
 
     ```bash
-    NODE_POD=$(kubectl -n services exec $OP_POD -c cray-console-operator -- sh -c \
-        "/app/get-node $XNAME" | jq .podname | sed 's/"//g')
-    echo $NODE_POD
+    WNODE=ncn-wxxx
     ```
 
-    Example output:
+1. (`ncn-mw#`) Cordon the node so that rescheduled pods do not end up back on the same node.
 
-    ```text
-    cray-console-node-2
+    ```bash
+    kubectl cordon $WNODE
     ```
 
-1. Find the worker node on which this pod is running.
+1. (`ncn-mw#`) Find all `cray-console` pods that need to be migrated.
+
+    This includes `cray-console-node`, `cray-console-data` (but not its Postgres pods), and `cray-console-operator`.
 
     ```bash
-    WNODE=$(kubectl get pods -o custom-columns=:.spec.nodeName -n services --no-headers $NODE_POD)
-    echo $WNODE
+    kubectl get pods -n services -l 'app.kubernetes.io/name in (cray-console-node, cray-console-data, cray-console-operator)' \
+      --field-selector spec.nodeName=$WNODE | awk '{print $1}'
     ```
 
     Example output:
 
     ```text
-    ncn-w003
+    cray-console-operator-6cf89ff566-kfnjr
     ```
 
-1. **Optional:** Move the `cray-console-node` pod.
+1. (`ncn-mw#`) Delete any `cray-console-operator` or `cray-console-data` pods listed in the previous step.
 
-    The pod can be proactively moved to a different worker if a power or reboot operation is going to be performed on the node where the pod is running.
+    If none were listed, then skip this step.
 
-    1. Prevent new pods from being scheduled on the NCN worker node currently running ConMan.
+    1. Delete the pods.
 
         ```bash
-        kubectl cordon $WNODE
+        for POD in $(kubectl get pods -n services -l 'app.kubernetes.io/name in (cray-console-data, cray-console-operator)' \
+          --field-selector spec.nodeName=$WNODE | awk '{print $1}'); do
+                kubectl -n services delete pod $POD
+        done
         ```
 
-    1. Delete the pod.
+    1. Wait for the `console-operator` and `console-data` pods to be re-scheduled on other nodes.
 
-        When the pod comes back up, it will be on a different NCN worker node.
+        Run the following command until both deployments show `1/1` pods are ready.
 
         ```bash
-        kubectl -n services delete $NODE_POD
+        kubectl -n services get deployment | grep cray-console
         ```
 
-    1. Wait for pod to terminate and come back up again.
+        Example output:
 
-        It may take several minutes even after the pod is running for the console connections to be re-established.
+        ```text
+        cray-console-data           1/1     1          1     103d
```

Comment:
Minor point - if these are newly moved they won't have a lifetime of 103d...

Reply:
Good point. Fixed

Reply:
Did you push up your changes? They don't seem to be showing up for me.